Ports/lite-xl/patches/0001-Add-SerenityOS-platform-support.patch

```diff
@@ -0,0 +1,144 @@
+From a5976576c24ebd870cb7d4533527b65ec0a5acfa Mon Sep 17 00:00:00 2001
+From: Jan200101 <sentrycraft123@gmail.com>
+Date: Mon, 3 Oct 2022 18:21:44 +0200
+Subject: [PATCH] Add SerenityOS platform support
+
+Signed-off-by: Jan200101 <sentrycraft123@gmail.com>
+---
+ meson.build                       |  6 +++-
+ src/api/dirmonitor/inodewatcher.c | 54 +++++++++++++++++++++++++++++++
+ src/main.c                        |  6 ++--
+ src/meson.build                   |  2 ++
+ 4 files changed, 65 insertions(+), 3 deletions(-)
+ create mode 100644 src/api/dirmonitor/inodewatcher.c
+
+diff --git a/meson.build b/meson.build
+index 9454fce..57c7722 100644
+--- a/meson.build
++++ b/meson.build
+@@ -82,13 +82,17 @@ if not get_option('source-only')
+ 
+     foreach lua : lua_names
+         last_lua = (lua == lua_names[-1] or get_option('wrap_mode') == 'forcefallback')
+-        lua_dep = dependency(lua, fallback: last_lua ? ['lua', 'lua_dep'] : [], required : last_lua,
++        lua_dep = dependency(lua, fallback: last_lua ? ['lua', 'lua_dep'] : [], required : false,
+             version: '>= 5.4',
+             default_options: default_fallback_options + ['default_library=static', 'line_editing=false', 'interpreter=false']
+         )
+         if lua_dep.found()
+             break
+         endif
++
++        if last_lua
++            lua_dep = cc.find_library('lua', required: true)
++        endif
+     endforeach
+ 
+     pcre2_dep = dependency('libpcre2-8', fallback: ['pcre2', 'libpcre2_8'],
+diff --git a/src/api/dirmonitor/inodewatcher.c b/src/api/dirmonitor/inodewatcher.c
+new file mode 100644
+index 0000000..8ca0b4f
+--- /dev/null
++++ b/src/api/dirmonitor/inodewatcher.c
+@@ -0,0 +1,54 @@
++#include <stdlib.h>
++#include <unistd.h>
++#include <fcntl.h>
++#include <poll.h>
++#include <string.h>
++#include <limits.h>
++
++// C version of InodeWatcherEvent
++struct [[gnu::packed]] C_InodeWatcherEvent {
++    int watch_descriptor;
++    uint32_t type;
++    size_t name_length;
++    // This is a VLA which is written during the read() from the descriptor.
++    char const name[];
++};
```

Comment:
This sounds like it breaks hard if we ever change `InodeWatcherEvent` down the line. Is there a chance that we can do some `#ifdef __cplusplus` magic to expose a C-only version directly from `Kernel/API/InodeWatcherEvent.h`?

Reply:
the enum class stuff can easily be `#ifdef __cplusplus`'d away but this would still leave the the default values and the use of the `Invalid` enum.

Thus far I haven't found a good solution for this.

On the ports side this struct could be slimmed down to just `watch_descriptor` since nothing other part of the struct is actually used.

Reply:
Spend a while trying to come up with a way of making `InodeWatcherEvent` work in C and I really don't think there is one without moving parts of the Serenity code into the C world.

I'll rewrite that part in C++ so it can make use of the struct directly